src/gui/connectionvalidator.cpp

```diff
@@ -286,13 +286,25 @@ void ConnectionValidator::slotUserFetched(const QJsonDocument &json)
     if (!displayName.isEmpty()) {
         _account->setDavDisplayName(displayName);
     }
+
+    auto capabilities = _account->capabilities();
+    // We should have received the capabilities by now. Check that assumption in a debug build. If
+    // it's not the case, the code below will assume that they are not available.
+    Q_ASSERT(capabilities.isValid());
+
+    // Ugh, #ifdefs are gross.
 #ifndef TOKEN_AUTH_ONLY
-    AvatarJob *job = new AvatarJob(_account, _account->davUser(), 128, this);
-    job->setTimeout(20 * 1000);
-    QObject::connect(job, &AvatarJob::avatarPixmap, this, &ConnectionValidator::slotAvatarImage);
-    job->start();
-#else
-    reportResult(Connected);
+    if (capabilities.isValid() && capabilities.avatarsAvailable()) {
+        AvatarJob *job = new AvatarJob(_account, _account->davUser(), 128, this);
+        job->setTimeout(20 * 1000);
+        QObject::connect(job, &AvatarJob::avatarPixmap, this, &ConnectionValidator::slotAvatarImage);
+        job->start();
+        // reportResult will be called when the avatar has been received by `slotAvatarImage`
+    } else {
+#endif
```

Comment:
maybe just move the opening brace out if the ifdef?

For 2.10 we might consider to remove `TOKEN_AUTH_ONLY` as far as I understand my predecessor it was hard work to build the client without ui components but I'm pretty sure its broken anyhow....